### DISCUSSION TITLE
how to copy to Plugin into the App Bundle

### DISCUSSION THREAD

**Discussion by andybin-dev** [tauri-apps/tauri#5980] [Q&A] [answered]
I want to copy a safari Extension App Plugin into the App Bundle, Is there any way do this?
![image](https://user-images.githubusercontent.com/13436526/210501000-164a0657-9364-4bf6-a1fc-f846a03808a6.png)

like copy files into Resource directory:
https://tauri.app/v1/guides/building/resources

**Comment by FabianLars** [comment, marked as answer]
The Resources feature is pretty much the only way to bundle additional files but you cannot specify where it saves the files in the bundle.

Also just to be sure, you're not trying to bundle the extension so that it can be used in the Tauri app itself right? Tauri does not use Safari so extensions won't work.

**Reply by andybin-dev** [reply]
> you're not trying to bundle the extension so that it can be used in the Tauri app itself right? Tauri does not use Safari so extensions won't work.

@FabianLars  no, I am not use this extension plugin in Tauri app. I need to install the Extension app to the Safari Browser App, and the way is include the Extension plugin app into Plugins directory.

**Reply by andybin-dev** [reply]
FYI: the other App Extensions also need copy into bundle's Plugins directory. like: `iMessage Extension`  `Photo Editing Extension` etc. (but only for Mac)

<img width="733" alt="image" src="https://user-images.githubusercontent.com/13436526/210535397-583d2c94-36fe-4e3d-8985-c41f13039c43.png">

**Reply by FabianLars** [reply]
Hmm, there was this PR https://github.com/tauri-apps/tauri/pull/4138 but they closed it :/

**Reply by althaffe** [reply]
@andybin-cn did you find a way to do this?

**Reply by althaffe** [reply]
Never mind. Found it: https://github.com/tauri-apps/tauri/pull/7798
